api/konnect/v1alpha1/konnect_cloudgateway_network_types.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+)
+
+func init() {
+	SchemeBuilder.Register(&KonnectCloudGatewayNetwork{}, &KonnectCloudGatewayNetworkList{})
+}
+
+// KonnectCloudGatewayNetwork is the Schema for the Konnect Network API.
+//
+// +genclient
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:object:generate=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Programmed",description="The Resource is Programmed on Konnect",type=string,JSONPath=`.status.conditions[?(@.type=='Programmed')].status`
+// +kubebuilder:printcolumn:name="State",description="The state the network is in",type=string,JSONPath=`.status.state`
+// +kubebuilder:printcolumn:name="ID",description="Konnect ID",type=string,JSONPath=`.status.id`
+// +kubebuilder:printcolumn:name="OrgID",description="Konnect Organization ID this resource belongs to.",type=string,JSONPath=`.status.organizationID`
+// +apireference:kgo:include
+// +kong:channels=gateway-operator
+type KonnectCloudGatewayNetwork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of KonnectCloudGatewayNetwork.
+	Spec KonnectCloudGatewayNetworkSpec `json:"spec,omitempty"`
+
+	// Status defines the observed state of KonnectCloudGatewayNetwork.
+	Status KonnectCloudGatewayNetworkStatus `json:"status,omitempty"`
+}
+
+// KonnectCloudGatewayNetworkSpec defines the desired state of KonnectCloudGatewayNetwork.
+//
+// +apireference:kgo:include
+type KonnectCloudGatewayNetworkSpec struct {
+	// KonnectCloudGatewayNetworkAPISpec defines the desired state of the network
+	// as specified in the Konnect API contract.
+	KonnectCloudGatewayNetworkAPISpec `json:",inline"`
+
+	// +kubebuilder:validation:Required
+	KonnectConfiguration KonnectConfiguration `json:"konnect"`
+}
+
+// KonnectCloudGatewayNetworkAPISpec defines the desired state of the network
+// as specified in the Konnect API contract.
+//
+// +kubebuilder:validation:XValidation:rule="self.name == oldSelf.name", message="Network name is immutable"
+// +kubebuilder:validation:XValidation:rule="self.cloud_gateway_provider_account_id == oldSelf.cloud_gateway_provider_account_id", message="Network cloud gateway provider account ID is immutable"
+// +kubebuilder:validation:XValidation:rule="self.region == oldSelf.region", message="Network region is immutable"
+// +kubebuilder:validation:XValidation:rule="self.availability_zones == oldSelf.availability_zones", message="Network availability zones are immutable"
+// +kubebuilder:validation:XValidation:rule="self.cidr_block == oldSelf.cidr_block", message="Network CIDR block is immutable"
+// +kubebuilder:validation:XValidation:rule="(!has(self.state) && !has(oldSelf.state)) || self.state == oldSelf.state", message="Network state is immutable"
```

Comment:
Maybe we should check the programmed condition as the immutability condition? I think it would make sense to allow changes in case the resource is not programmed in Konnect yet. WDYT?

Reply:
Makes sense, let me address that.

Reply:
For some reason the validation rules added in https://github.com/Kong/kubernetes-configuration/pull/268/files#diff-4a7dbeb1a5090bccd221cc8c4541d55f93895a41ab977c1fb0aafee014f7a308R20-R27

are causing failures: https://github.com/Kong/kubernetes-configuration/actions/runs/13305937028/job/37156809821?pr=268#step:7:1876

```
KonnectCloudGatewayNetwork.konnect.konghq.com "test-cl8mh" is invalid: [<nil>: Invalid value: "object": no such key: status evaluating rule: spec.name is immutable when an entity is already Programmed,
...
```

🤔

Reply:
Ok, fixed with 09122417baf68cabe9d0dc80c575dda64c7a4a3a PTAL @mlavacca